MBBSEmu/HostProcess/GlobalRoutines/InstantMessageGlobal.cs

```diff
@@ -0,0 +1,51 @@
+﻿using MBBSEmu.Extensions;
+using MBBSEmu.Memory;
+using MBBSEmu.Module;
+using MBBSEmu.Session;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace MBBSEmu.HostProcess.GlobalRoutines
+{
+    public class InstantMessageGlobal : IGlobalRoutine
+    {
+        public bool ProcessCommand(byte[] command, ushort channelNumber, PointerDictionary<SessionBase> sessions, Dictionary<string, MbbsModule> modules)
+        {
+            var commandString = Encoding.ASCII.GetString(command).TrimEnd('\0');
+            
+            //Check minimum length of a /P command (one character username/one character message = 6) and make sure there are at least 2 spaces
+            if (commandString.Length > 6 && commandString.Substring(0,2).ToUpper() == "/P" && Regex.Matches(commandString, " ").Count > 1)
+            {
+                //Determine location of first 2 spaces to separate username and message
+                var instantMessageSpace1 = commandString.IndexOf(' ');
```

Comment:
It might be easier to `commandString.Split(' ')` to split up the individual words, getting `[1]` for the username then doing a `string.Join` on the remaining elements to get the message.